sormas-ui/src/main/java/de/symeda/sormas/ui/location/LocationEditForm.java

```diff
@@ -524,19 +525,18 @@ private HorizontalLayout createGeoButton() {
 			e.getButton().removeStyleName(CssStyles.GEOCODE_BUTTON_HIGHLIGHT);
 		}, ValoTheme.BUTTON_ICON_ONLY, ValoTheme.BUTTON_BORDERLESS, ValoTheme.BUTTON_LARGE);
 
+		Field[] locationGeoFields = Stream.of(LocationDto.STREET, LocationDto.POSTAL_CODE, LocationDto.CITY, LocationDto.HOUSE_NUMBER)
+			.map(field -> (Field) getField(field))
+			.toArray(Field[]::new);
+
 		// Highlight geocode-button when the address changes
-		for (String fieldName : new String[] {
-			LocationDto.STREET,
-			LocationDto.POSTAL_CODE,
-			LocationDto.CITY,
-			LocationDto.HOUSE_NUMBER }) {
-			Field<?> field = getField(fieldName);
-			field.addValueChangeListener(e -> {
-				if (field.isModified()) {
-					geocodeButton.addStyleName(CssStyles.GEOCODE_BUTTON_HIGHLIGHT);
-				}
-			});
-		} ;
+		Stream.of(locationGeoFields).forEach(field -> field.addValueChangeListener(e -> {
+			if (isAllFieldsEmpty(locationGeoFields)) {
+				geocodeButton.removeStyleName(CssStyles.GEOCODE_BUTTON_HIGHLIGHT);
+			} else if (field.isModified()) {
```

Comment:
Still not changed as I commented in Crowd testing: `if (field.isModified())` changes behaviour. Intended was to signal the user to click on geocoding button when the value changed. Now when you change a value (`modified = true`), click on geo button, and change than the value back as it is in the DTO (`modified = false`), the highlighting does not reappear. I would say: Just remove the if clause: If the value in any related text field changes, highlight the geocoding button as it was before.

Reply:
Tested with @syntakker: Is okay, isModified does not change to false if the DTO value is typed in again. So my objection is not relevant.